functions/src/definitions/eventHandlers/checkerHandlerTelegram.ts

```diff
@@ -146,4 +201,349 @@ const onCheckerPublishTelegram = onMessagePublished(
+  await bot.sendMessage(
+    chatId,
+    "Thank you for adding the WA bot. Please join the CheckMate Checker's groupchat: https://t.me/CheckMate_Checker_Bot.", //UPDATE the groupchat link here
+    {
+      reply_markup: {
+        inline_keyboard: [
+          [
+            {
+              text: "Yes, I have added the telegram bot",
+              callback_data: "TG_COMPLETED",
+            },
+          ],
+        ],
+      },
+    }
+  )
+}
+
+const sendCompletionPrompt = async (chatId: number) => {
+  await bot.sendMessage(
+    chatId,
+    "Thank you for adding the telegram bot. Please refer to CheckMate wiki for more information: https://checkmate.sg/."
+  )
+
+  await bot.sendMessage(
+    chatId,
+    `You have successfully onboarded as a CheckMate Checker!`
+  )
+}
+
+//checks for sendWAGroupPrompt / sendTGGroupPrompt / sendCompletionPrompt responses + FUA
+bot.on("callback_query", async function onCallbackQuery(callbackQuery) {
+  const action = callbackQuery.data
+  const chatId = callbackQuery.message?.chat.id
+  if (chatId != undefined) {
+    const checkerDocQuery = db
+      .collection("checkers")
+      .where("telegramId", "==", callbackQuery.from.id)
+    const checkerQuerySnap = await checkerDocQuery.get()
+
+    switch (action) {
+      case "QUIZ_COMPLETED":
+        if (checkerQuerySnap.docs[0].data()?.onboardingStatus === "waGroup") {
+          sendWAGroupPrompt(chatId)
+        } else {
+          await bot.sendMessage(chatId, `Please complete the Onboarding quiz.`)
+        }
+
+        break
+      case "WA_COMPLETED":
+        // check WA bot completion
+        const whatsappId = checkerQuerySnap.docs[0].data()?.whatsappId
+        const userSnap = await db.collection("users").doc(whatsappId).get()
+
+        if (userSnap.exists) {
+          await checkerQuerySnap.docs[0].ref.update({
+            onboardingStatus: "tgGroup",
+          })
+
+          sendTGGroupPrompt(chatId)
+        } else {
+          await bot.sendMessage(
+            chatId,
+            `Please add the CheckMate Whatsapp bot: https://wa.me/6580432188.`,
+            {
+              reply_markup: {
+                inline_keyboard: [
+                  [
+                    {
+                      text: "Yes, I have added the WA bot",
+                      callback_data: "WA_COMPLETED",
+                    },
+                  ],
+                ],
+              },
+            }
+          )
+        }
+        break
+      case "TG_COMPLETED":
+        // check tele bot completion
+        const member = await bot.getChatMember(
+          CHECKERS_CHAT_ID,
+          callbackQuery.from.id
+        )
+        if (member.status == "member") {
+          await checkerQuerySnap.docs[0].ref.update({
+            onboardingStatus: "completed",
+            isOnboardingComplete: true,
+          })
+
+          sendCompletionPrompt(chatId)
+        } else {
+          await bot.sendMessage(
```

Comment:
Suggest add a flag to sendTGGroupPrompt to handle the case where it's a re-prompt. Then adjust the prefix based on the flag. So like if re-prompt, prefix would be 'We noticed you havent joined yet. ' if not, then it'd be 'Thank you for adding the WA bot. '